paging/paging-common/src/test/kotlin/androidx/paging/PagingDataDifferTest.kt

```diff
@@ -163,6 +163,339 @@ class PagingDataDifferTest(
         job.cancel()
     }
 
+    @Test
+    fun uiReceiverSetImmediately() = testScope.runTest {
+        val differ = SimpleDiffer(differCallback = dummyDifferCallback)
+        val receiver = UiReceiverFake()
+        val pagingData1 = infinitelySuspendingPagingData(uiReceiver = receiver)
+
+        val job1 = launch {
+            differ.collectFrom(pagingData1)
+        }
+        assertTrue(job1.isActive) // ensure job started
+
+        assertThat(receiver.refreshEvents).hasSize(0)
+
+        differ.refresh()
+        // double check that the pagingdata's receiver was registered and had received refresh call
+        // before any PageEvent is collected/presented
+        assertThat(receiver.refreshEvents).hasSize(1)
+
+        job1.cancel()
+    }
+
+    @Test
+    fun hintReceiverSetAfterNewListPresented() = testScope.runTest {
+        val differ = SimpleDiffer(differCallback = dummyDifferCallback)
+
+        // first generation, load something so next gen can access index to trigger hint
+        val hintReceiver1 = HintReceiverFake()
+        val flow = flowOf(
+            localRefresh(pages = listOf(TransformablePage(listOf(0, 1, 2, 3, 4)))),
+        )
+
+        val job1 = launch {
+            differ.collectFrom(PagingData(flow, dummyUiReceiver, hintReceiver1))
+        }
+        assertThat(hintReceiver1.hints).hasSize(1) // initial hint
+
+        // trigger second generation
+        differ.refresh()
+
+        // second generation
+        val pageEventCh = Channel<PageEvent<Int>>(Channel.UNLIMITED)
+        val hintReceiver2 = HintReceiverFake()
+        val job2 = launch {
+            differ.collectFrom(
+                PagingData(pageEventCh.consumeAsFlow(), dummyUiReceiver, hintReceiver2)
+            )
+        }
+
+        // we send the initial load state. this should NOT cause second gen receiver to register
```

Comment:
This should be `second gen hint receiver to register`, correct?

Reply:
done